src/wayland-eglsurface.c

```diff
@@ -2683,6 +2685,23 @@ EGLSurface wlEglCreatePlatformWindowSurfaceHook(EGLDisplay dpy,
         goto fail;
     }
 
+    getWlEglWindowVersionAndSurface(window, &wver, &wsurf);
+    if (wsurf == NULL) {
+        err = EGL_BAD_ALLOC;
+        goto fail;
+    }
```

Comment:
Here we store our wl_surface in `wsurf` so we don't prematurely set it, but doesn't this leak the surface if we jump to `fail` later on (like in line 2708? Before this change I think this would get cleaned up as part of `wlEglDestroySurface` but now this isn't recorded in `surface` until later?

I think we can go ahead and set `surface->wlSurface` here? Afaict that shouldn't affect anything between here and 2779 and properly handles the surface cleanup.

Reply:
`getWlEglWindowVersionAndSurface` doesn't allocate anything, it just figures out the correct offset for the `wl_surface` pointer in the `wl_egl_surface` struct.

That said, using the same `surface` variable in the loop *would* cause problems in the failure path, because it'll try to clean up an existing `WlEglSurface`.

Reply:
Ah good point, thanks that looks better now.